src/couch_ssl/src/couch_ssl.erl

```diff
@@ -0,0 +1,49 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_ssl).
+
+-export([childspecs/0, listen/1, accept/1, accept_connection/5,
+    setup/5, close/1, select/1, is_node_name/1]).
+
+childspecs() ->
+    {ok, [{ssl_dist_sup, {ssl_dist_sup, start_link, []},
+        permanent, infinity, supervisor, [ssl_dist_sup]}]}.
+
+listen(Name) ->
+    inet_tls_dist:listen(Name).
+
+
+accept(Listen) ->
+    inet_tls_dist:accept(Listen).
+
+
+accept_connection(AcceptPid, Socket, MyNode, Allowed, SetupTime) ->
+    inet_tcp_dist:accept_connection(
```

Comment:
I am a bit confused how this is supposed to work? When creating the listen socket we call into the TSL dist module but when handling the accepted socket we use TCP?

Reply:
Changed to `inet_tls_dist`. Thanks for the review, @nickva